library/src/main/java/com/google/maps/android/data/kml/KmlLayer.java

```diff
@@ -106,14 +120,48 @@ public KmlLayer(GoogleMap map, InputStream stream, FragmentActivity activity, Ma
         if (stream == null) {
             throw new IllegalArgumentException("KML InputStream cannot be null");
         }
-        KmlRenderer mRenderer = new KmlRenderer(map, activity, markerManager, polygonManager, polylineManager, groundOverlayManager);
+        KmlRenderer renderer = new KmlRenderer(map, activity, markerManager, polygonManager, polylineManager, groundOverlayManager);
+
+        BufferedInputStream bis = new BufferedInputStream(stream);
+        bis.mark(1024);
+        ZipInputStream zip = new ZipInputStream(bis);
+        try {
+            KmlParser parser = null;
+            ZipEntry entry = zip.getNextEntry();
+            if (entry != null) { // is a KMZ zip file
+                while (entry != null) {
+                    if (parser == null && entry.getName().toLowerCase().endsWith(".kml")) {
+                        parser = parseKml(zip);
+                    } else {
+                        Bitmap bitmap = BitmapFactory.decodeStream(zip);
+                        if (bitmap != null) {
+                            renderer.cacheBitmap(entry.getName(), bitmap);
```

Comment:
Instead of caching the bitmap in `KmlLayer`, a helper method in `KmlRenderer` would be cleaner:

e.g. 
```java
KmlRenderer.storeKmzData(HashMap<String, KmlStyle> styles,
    HashMap<String, String> styleMaps,
    HashMap<KmlPlacemark, Object> features, ArrayList<KmlContainer> folders,
    HashMap<KmlGroundOverlay, GroundOverlay> groundOverlays,
    Set<String> bitmapInputStreams)
```

This way, all bitmap caching is done within `KmlRenderer` and the scoping of the functions within `Renderer` can be reverted.

Reply:
I added a similar method. It's not possible to pass a set of bitmap `InputStream`s, as each bitmap uses the same `ZipInputStream` as the `ZipEntry`s are iterated over. So I'm just passing a map of decoded bitmaps. https://github.com/googlemaps/android-maps-utils/pull/625/commits/482c748605b3d9a46a88118b1a9304d3bf77644d